modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/discovery/DataDiscovery.scala

```diff
@@ -167,26 +184,20 @@ object DataDiscovery {
 
   def getShredModels[F[_]: Monad: Iglu](
     nonAtomicTypes: List[ShreddedType]
-  ): EitherT[F, LoaderError, Map[SchemaKey, MergeRedshiftSchemasResult]] = {
-    val maxSchemaKeyPerTableName = getMaxSchemaKeyPerTableName(nonAtomicTypes)
+  ): EitherT[F, LoaderError, Map[SchemaKey, DiscoveredShredModels]] =
     nonAtomicTypes
       .traverse { shreddedType =>
         EitherT(Iglu[F].getSchemasWithSameModel(shreddedType.info.getSchemaKey)).map { schemas =>
-          val maxSchemaKey = maxSchemaKeyPerTableName(shreddedType.info.getName)
-          val filtered = schemas.filter(_.self.schemaKey <= maxSchemaKey).toNel.get
-          val mergeRedshiftSchemasResult = foldMapMergeRedshiftSchemas(filtered)
-          (shreddedType.info.getSchemaKey, mergeRedshiftSchemasResult)
+          val filtered = schemas.filter(_.self.schemaKey <= shreddedType.info.getSchemaKey).toNel.get
+          val foldMapRedshiftSchemasResult: ShredModel = foldMapRedshiftSchemas(filtered)(shreddedType.info.getSchemaKey)
+          val foldMapMergeRedshiftSchemasResult = foldMapMergeRedshiftSchemas(filtered)
```

Comment:
Let's dig into this line, where you call `foldMapMergeRedshiftSchemas` with a filtered list.

Before this change, the filtered list was filtered based on the value from `maxSchemaKeyPerTableName`.  But after this change, the filtered list is filtered based on `shreddedType.info.getSchemaKey`.  So now, if shredded types are similar but not identical (e.g. 1-0-1 and 1-0-2), then you return a different merge result for each.  Whereas before it returned the same merge result for each.

This then makes a difference on [this line](https://github.com/snowplow/snowplow-rdb-loader/blob/02d7bafd8a6589d3e59cb3bacd7115bdbee4481f/modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/db/Migration.scala#L180) where you call `.toSet` on the merge results.

I'm not saying it's a problem -- I'm just pointing out that there's a difference and you might not have spotted it!

Can you envisage batches where the result of `descriptionList.toSet` will be different after this change?  And does that cause any problems for applying migrations?

Reply:
reverted to the previous logic to construct shred models for migration